src/Microsoft.ServiceFabric.Actors/Runtime/KvsActorStateProvider.cs

```diff
@@ -166,17 +175,113 @@ private LocalStoreSettings GetLocalStoreSettings()
 
         private KeyValueStoreReplicaSettings GetKvsReplicaSettings()
         {
-            var kvsReplicaSettings = new KeyValueStoreReplicaSettings
+            if (this.userDefinedKeyValueStoreReplicaSettings == null)
             {
-                SecondaryNotificationMode = KeyValueStoreReplica.SecondaryNotificationMode.NonBlockingQuorumAcked,
-            };
+                var configPackageName = ActorNameFormat.GetConfigPackageName(this.ActorTypeInformation.ImplementationType);
+                var kvsSettingsSectionName = ActorNameFormat.GetKeyValueStoreSettingsConfigSectionName(this.ActorTypeInformation.ImplementationType);
+                var configPackageObj = this.InitParams.CodePackageActivationContext.GetConfigurationPackageObject(configPackageName);
+
+                KeyValueStoreReplicaSettings kvsReplicaSettings;
+                if (configPackageObj.Settings.Sections.Contains(kvsSettingsSectionName))
+                {
+                    kvsReplicaSettings = KeyValueStoreReplicaSettingsLoader.LoadFrom(
+                        this.InitParams.CodePackageActivationContext,
+                        configPackageName,
+                        kvsSettingsSectionName);
+                }
+                else
+                {
+                    kvsReplicaSettings = new KeyValueStoreReplicaSettings()
+                    {
+                        SecondaryNotificationMode = SecondaryNotificationMode.NonBlockingQuorumAcked,
+                    };
+                }
 
-            if (this.userDefinedLogTruncationInterval.HasValue)
+                if (this.userDefinedLogTruncationInterval.HasValue)
+                {
+                    kvsReplicaSettings.LogTruncationIntervalInMinutes = this.userDefinedLogTruncationInterval.Value;
+                }
+
+                return kvsReplicaSettings;
+            }
+
+            return this.userDefinedKeyValueStoreReplicaSettings;
+        }
+
+        private static class KeyValueStoreReplicaSettingsLoader
+        {
+            public static KeyValueStoreReplicaSettings LoadFrom(
+                CodePackageActivationContext codePackageActivationContext,
+                string configPackageName,
+                string sectionName)
             {
-                kvsReplicaSettings.LogTruncationIntervalInMinutes = this.userDefinedLogTruncationInterval.Value;
+                var kvsSettings = new KeyValueStoreReplicaSettings();
+                var configPackage = codePackageActivationContext.GetConfigurationPackageObject(configPackageName);
+                if (configPackage.Settings.Sections.Contains(sectionName))
+                {
+                    var section = configPackage.Settings.Sections[sectionName];
+
+                    string value;
+                    if (TryReadSettingValue(section, "TransactionDrainTimeout", out value))

```

Comment:
I thought we will parse config in runtime and will use `::LoadFrom` fn here . Has that changed ?